dbms/src/TestUtils/ExecutorTestUtils.cpp

```diff
@@ -211,40 +207,37 @@ void ExecutorTest::enablePlanner(bool is_enable)
 
 DB::ColumnsWithTypeAndName ExecutorTest::executeStreams(
     const std::shared_ptr<tipb::DAGRequest> & request,
-    size_t concurrency)
+    size_t concurrency,
+    bool enable_memory_tracker)
 {
     DAGContext dag_context(*request, "executor_test", concurrency);
-    return executeStreams(&dag_context);
+    return executeStreams(&dag_context, enable_memory_tracker);
 }
 
-ColumnsWithTypeAndName ExecutorTest::executeStreams(DAGContext * dag_context)
+ColumnsWithTypeAndName ExecutorTest::executeStreams(DAGContext * dag_context, bool enable_memory_tracker)
 {
-    context.context.setExecutorTest();
-    context.context.setMockStorage(context.mockStorage());
-    context.context.setDAGContext(dag_context);
+    TiFlashTestEnv::setUpTestContext(context.context, dag_context, context.mockStorage(), TestType::EXECUTOR_TEST);
     // Currently, don't care about regions information in tests.
     Blocks blocks;
-    queryExecute(context.context, /*internal=*/true)->execute([&blocks](const Block & block) { blocks.push_back(block); }).verify();
+    queryExecute(context.context, /*internal=*/!enable_memory_tracker)->execute([&blocks](const Block & block) { blocks.push_back(block); }).verify();
     return mergeBlocks(std::move(blocks)).getColumnsWithTypeAndName();
 }
 
-Blocks ExecutorTest::getExecuteStreamsReturnBlocks(const std::shared_ptr<tipb::DAGRequest> & request, size_t concurrency)
+Blocks ExecutorTest::getExecuteStreamsReturnBlocks(const std::shared_ptr<tipb::DAGRequest> & request,
+                                                   size_t concurrency,
+                                                   bool enable_memory_tracker)
 {
     DAGContext dag_context(*request, "executor_test", concurrency);
-    context.context.setExecutorTest();
-    context.context.setMockStorage(context.mockStorage());
-    context.context.setDAGContext(&dag_context);
+    TiFlashTestEnv::setUpTestContext(context.context, &dag_context, context.mockStorage(), TestType::EXECUTOR_TEST);
     // Currently, don't care about regions information in tests.
     Blocks blocks;
-    queryExecute(context.context, /*internal=*/true)->execute([&blocks](const Block & block) { blocks.push_back(block); }).verify();
+    queryExecute(context.context, /*internal=*/!enable_memory_tracker)->execute([&blocks](const Block & block) { blocks.push_back(block); }).verify();
```

Comment:
maybe we can remove `internal` and enable memory tracker in all case in next pr.

Reply:
Yes, I aggree.